Add product card styling.

madeincosmos opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:
- Update product cards on the WooCommerce > Extensions page in WP Admin to match current designs.
<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

### How to test the changes in this Pull Request:

1. Check out this branch on local environment.
2. Navigate to one of marketplace category pages, i.e. `wp-admin/admin.php?page=wc-addons&section=payment-gateways`  (nav menu isn't implemented at the moment
3. Check product card and make sure it looks good on all screen sizes

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable? - e2e tests are currently failing because the WooCommerce.com Subscriptions screen is now in a different place, these changes are addressed in #30380.
* [x] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Screenshots

![](https://d.pr/i/zpIwTT+)

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

madeincosmos committed:
Add product card styling.

andfinally reviewed (commented):
Looking great, thanks! There are a few things I still need to check. I think we probably need to retain the max-width 1200px on the list of products and align it left, like in the current page. So we may want to adjust the width of the cards a little bit to allow for that greater width.

madeincosmos committed:
Align product cards on the Extensions page.

madeincosmos commented on assets/css/admin.scss:
```diff
@@ -537,72 +537,85 @@
-				inset 0 -1px 0 rgba(0, 0, 0, 0.1);
+			padding: 0;
+			vertical-align: top;
+			width: 45%;
```
Added max width on product card and an extra wrapper to make the product list look good on all screen sizes:

![](https://d.pr/i/7ozQGf+)

tgglv commented:
Hey @madeincosmos, just started to see this PR and noticed that E2E tests [failed](https://github.com/woocommerce/woocommerce/pull/30410/checks?check_run_id=3278191858):
```
a.nav-tab (text: "WooCommerce.com Subscriptions") not found
```
It looks like the expected element is missed. It might happen coincidentally. Can you take a look?

madeincosmos commented:
> Hey @madeincosmos, just started to see this PR and noticed that E2E tests [failed](https://github.com/woocommerce/woocommerce/pull/30410/checks?check_run_id=3278191858):

Hi @tgglv, the e2e tests are failing because we changed the marketplace page structure. I've updated the tests in a different branch that is currently merged into ` update/marketplace-1`, I can try to rebase this one if that's needed.

andfinally committed:
Merge branch 'update/marketplace-1' into update/styling-product-cards

andfinally committed:
CSS and layout tweaks.

- Changed `wc_addons_wrap` to `wc-addons-wrap` to adhere to stylelint class rule.
- Removed `max-width` from `wc-addons-wrap` – we need the purple header to stretch to full width. Added new `wc-subscriptions-wrap` class to give the `max-width` to the My Subscriptions page.
- Changed width of product cards to use `calc`, so that any trailing odd-numbered card has the same width as the others.
- Fixed some stylelint issues. Moved style blocks to address `no-descending-specificity` warnings. Remove quotes from URLs and font names that don't need them.
- Gave `#screen-meta-links` absolute position on the marketplace page, so it overlays the full-width purple header.

andfinally committed:
Addressing PHPCS errors.

- Deleted precision indentation on lines 72 and 80.
- Deleted unnecessary `echo` from before `esc_html_e` cals.

andfinally reviewed (approved):
Looks lovely! My apologies Maria, I pushed some additions to this PR before I saw your [comment](https://github.com/woocommerce/woocommerce/pull/30410#discussion_r683270518), hope this is consistent with your work. I made some tweaks to the page layout.

- Made the wrapper full width, so the purple header stretches across the screen.
- Aligned the products to the left.
- Used `calc` for product card width, so any trailing odd-numbered card keeps the same width as the others.
- Addressed some stylelint warnings in `admin.scss`.

The PHP7.4 WP Nightly test is repeatedly failing, something to do with composer dependencies – I don't believe it's related to the changes in this PR.

madeincosmos committed:
Add product list margin on larger screens.

madeincosmos commented:
Thanks @andfinally, I like your approach better than the previous calculations based on margins. I can see in the designs that there's 84px space between WordPress menu and product list, so I've added this on larger screens. Other than this, I think we're good to.
